src/main/groovy/io/seqera/wave/controller/MetricsController.groovy

```diff
@@ -0,0 +1,168 @@
+/*
+ *  Wave, containers provisioning service
+ *  Copyright (c) 2024, Seqera Labs
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+package io.seqera.wave.controller
+
+import java.time.Instant
+import java.time.LocalDate
+import java.time.LocalTime
+import java.time.ZoneId
+import java.time.format.DateTimeParseException
+import javax.annotation.Nullable
+
+import groovy.transform.CompileStatic
+import groovy.util.logging.Slf4j
+import io.micronaut.http.HttpResponse
+import io.micronaut.http.MediaType
+import io.micronaut.http.annotation.Controller
+import io.micronaut.http.annotation.Error
+import io.micronaut.http.annotation.Get
+import io.micronaut.http.annotation.PathVariable
+import io.micronaut.http.annotation.QueryValue
+import io.micronaut.security.annotation.Secured
+import io.micronaut.security.authentication.AuthorizationException
+import io.micronaut.security.rules.SecurityRule
+import io.seqera.wave.service.metric.Metric
+import io.seqera.wave.service.metric.MetricFilter
+import io.seqera.wave.service.metric.MetricService
+import jakarta.inject.Inject
+import static io.micronaut.http.HttpHeaders.WWW_AUTHENTICATE
+
+/**
+ * Controller for wave Metrics
+ *
+ * @author Munish Chouhan <munish.chouhan@seqera.io>
+ */
+@Slf4j
+@CompileStatic
+@Secured(SecurityRule.IS_AUTHENTICATED)
+@Controller("/v1alpha1/metrics")
+class MetricsController {
+    @Inject
+    private MetricService metricsService
+
+    @Get(uri = "/builds/{metric}", produces = MediaType.APPLICATION_JSON)
+    HttpResponse<Map> getBuildMetrics(@PathVariable String metric,
+                                      @Nullable @QueryValue Boolean success,
+                                      @Nullable @QueryValue String startDate,
+                                      @Nullable @QueryValue String endDate,
+                                      @Nullable @QueryValue Integer limit) {
+        return HttpResponse.ok(
+                metricsService.getBuildMetrics(
+                        Metric.valueOf(metric),
+                        new MetricFilter.Builder()
+                                .dates(parseStartDate(startDate), parseEndDate(endDate))
+                                .success(success)
+                                .limit(limit)
```

Comment:
Also, `limit` should have a max constraint e.g. no more than 1000 entries

Reply:
I have added 
minimum limit = 1
default limit = 100 
and maximum limit = 1000

https://github.com/seqeralabs/wave/blob/9bde9fbcb748679ce1ec2871c7f19c2b823ab165/src/main/groovy/io/seqera/wave/service/metric/MetricFilter.groovy#L92-L95